conpot/core/protocol_wrapper.py

```diff
@@ -0,0 +1,29 @@
+from conpot.core import get_interface
+from datetime import datetime
+
+core_interface = get_interface()
+
+
+def conpot_protocol(cls):
+    class Wrapper(object):
+        def __init__(self, *args, **kwargs):
+            self.wrapped = cls(*args, **kwargs)
+            self.cls = cls
+            if self.cls.__name__ not in 'Proxy':
+                core_interface.protocols[self.cls] = self.wrapped
+
+        def __getattr__(self, name):
+            if name == 'handle':
+                # assuming that handle function from a class is only called when a client tries to connect with an
+                # enabled protocol, update the last_active (last_attacked attribute)
+                # FIXME: No handle function in HTTPServer
+                core_interface.last_active = datetime.now().strftime("%b %d %Y - %H:%M:%S")
+            return getattr(core_interface.protocols[self.cls], name)
+
+        def __repr__(self):
+            return self.cls.__repr__(self.wrapped)
+
+        __doc__ = property(lambda self: self.cls.__doc__)
```

Comment:
Well the downside of this approach is - every object that we create in `bin/conpot` is an instance this `Wrapper` class and no longer the instance of the actual protocol class. As you can see, I am monkey-patching the important stuff. 
So too many changes here and there could lead to very unexpected results. 

However, I would like to argue that unplugging a protocol from the internal interface is as easy as removing the `@conpot_protocol` decorator. Everything would still run perfect.

Reply:
For now I see the benefits and I am really curious how this will affect subsequent development (in terms of making things more flexible and easier to tie together).

A last question on this matter: How do you (personally) feel about debugging? Unexpected results is one thing - but I am a little unsure if this will have negative impact on how straight forward debugging issues will be in future :) So if somebody has any strong opinions on this specific matter, I'm open for input :)

Reply:
My objective for internal interface has always been to let protocols interact more deeply with each other. Not just for debugging. [So for example if users want an option to change databus values while Conpot is still running](https://github.com/mushorg/conpot/issues/372), they can do so - if we extend the interface ;-)

IMO as long we **maintain** a plug/unplug mechanism, there is little to worry about. We can always go for - 'X' protocols with interface and 'Y' protocols without it. Conpot would still run all protocols like nothing has changed.